src/random.cpp

```diff
@@ -36,9 +36,13 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/aux_/disable_warnings_push.hpp"
 
+#ifdef BOOST_NO_CXX11_HDR_RANDOM
```

Comment:
I don't know if `BOOST_NO_CXX11_HDR_RANDOM` is reliable. It is part of Boost.Config which you already use. I had thought of just checking if `__cplusplus` is c++11.
I have no preference. Whatever you tell me to put here I'll do it.